rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala

```diff
@@ -39,33 +39,57 @@ package com.normation.rudder.datasources
 
 import org.joda.time.DateTime
 import org.joda.time.Seconds
+import net.liftweb.common._
 
 sealed trait DataSourceType {
   def name : String
```

Comment:
And (because that's the last comment ;) I believe we also need to expose a "datasource update" timeout, which is a global timeout for the update (can be infinite) 
IE, for ex "I don't want the task to ever take more than 5 minutes, because it's a problem on my set up that need correction" or "I don't care, just get the data even if it means wainting 20 minutes on each generation".